### PR TITLE
Hooks: add cryptodome.

### PR DESCRIPTION
**Pull request by xoviat**

### CODE REVIEW

File: PyInstaller/hooks/hook-Cryptodome.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+binaries = []
+binary_module_names = (
+    'Cryptodome.Cipher',
+    'Cryptodome.Util',
+    'Cryptodome.Hash',
+)
+
+for module_path in binary_module_names:
+    m_dir = os.path.dirname(get_module_file_attribute(module_path))
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-Cryptodome.py]: Sorry for nit-picking, bit this is confusion for me: `…path in …names` and also `get_module_…(…path)`. Can you please rename `module_path` into `module_name`

File: PyInstaller/hooks/hook-Cryptodome.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    for ext in EXTENSION_SUFFIXES:
+        module_bin = glob.glob(os.path.join(m_dir, '_*%s' % ext))
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-Cryptodome.py]: I assume this includes all dynamic libraries starting with an underscore?

File: PyInstaller/hooks/hook-Cryptodome.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        for f in module_bin:
+            binaries.append((f, module_path.replace('.', '/')))
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-Cryptodome.py]: I assume, the `/` ought to be os.sep?

File: PyInstaller/hooks/hook-Cryptodome.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import os.path
+import glob
+
+from PyInstaller.compat import EXTENSION_SUFFIXES
+from PyInstaller.utils.hooks import get_module_file_attribute
+
+# Include the modules as binaries in a subfolder named like the package.
+# Cryptodome's loader expects to find them inside the package directory for
+# the main module. We cannot use hiddenimports because that would add the
+# modules outside the package.
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-Cryptodome.py]: But can't we use `find_subpackages` (or something like this) to find the modules?

**Comment by xoviat** [comment]
This is a duplicate of #2226 with the commit messages adjusted.

File: PyInstaller/hooks/hook-Cryptodome.py (hunk 3 above)
Comment (xoviat) [inline PyInstaller/hooks/hook-Cryptodome.py]: I can make these changes if you want, but you're guess is good as mine here.

Comment (htgoebel) [inline PyInstaller/hooks/hook-Cryptodome.py]: I'll take care of it.

**Comment by htgoebel** [comment]
So maybe its time for this old hint :-)

When updating a pull-request, you can simply (force) push the updated branch to github again. This will automatically update the pull-request (which follows the branch, not the commit). So you do not need to close the pull-request and open a new one. This also has the benefit that the discussion history is kept.

Please also take the change and rebase your branch on the current `develop` head. Thanks.

**Comment by xoviat** [comment]
> So maybe its time for this old hint :-)

It's not my PR. It's someone else's PR that I took over.

**Comment by htgoebel** [comment]
Oh, it time for bed. I thought it's been stephanrauchs PR, since the commits are authored by him. So my comment was addressed to stefan (from whom I thought, he'd know this). Just forget my comment.

File: PyInstaller/hooks/hook-Cryptodome.py (hunk 3 above)
Comment (htgoebel) [inline PyInstaller/hooks/hook-Cryptodome.py]: Sorry, I meant: I'll take care of the `os,sep`, not of the remaining changes :-)

Comment (xoviat) [inline PyInstaller/hooks/hook-Cryptodome.py]: Okay then. I'll work on this after modulegraph is sorted.

File: PyInstaller/hooks/hook-Cryptodome.py (hunk 4 above)
Comment (xoviat) [inline PyInstaller/hooks/hook-Cryptodome.py]: Where is `find_subpackages`?

Comment (htgoebel) [inline PyInstaller/hooks/hook-Cryptodome.py]: It's called `collect_submodules()`:
from PyInstaller.utils.hooks import collect_submodules

**Pull request merged**